### ISSUE TITLE
what should be a recommended value for signif_level for real data

### ISSUE THREAD

**Issue by smilesun** [agisga/ThresholdoutAUC#5]
when set `signif_level = 0`,  it is computed faster, but a bigger signif_level runs too slow, what would be a good recommended value, is `signif_level = 0.001` ok for real data?

**Comment by agisga** [comment]
`signif_level` determines which features are considered as candidate variables to be used for classification. It's used in the function [`fit_model_for_every_subset`](https://github.com/DIDSR/ThresholdoutAUC/blob/master/functions/fit_model_for_every_subset.R#L57). To be more precise, "candidate variables" are determined by performing a 2-sample t-test with significance level `signif_level` between every feature and the class label.

That is, if `signif_level = 1` (or very large), then a classifier will be fit to every possible combination of features, which will result in 2^p model fits if there are p features available (i.e., will take a really long time). If `signif_level = 0`, then only a *single* classification model is fit using the 2 most "significant" predictors only. If `signif_level > 0` but small, then only a handful of models will be fit.

That type of model selection on training data happens within every "round of adaptive learning".

**Comment by agisga** [comment]
I used `signif_level = 0.01`, as you can see here: https://github.com/DIDSR/ThresholdoutAUC/blob/master/set_params.R
That was sufficiently small/large for my purposes of studying overfitting to a test dataset that is re-used repeatedly.
What parameter values will be okay for your use case will depend on your purpose for performing these computational experiments (which I don't know what it is).
If you set `signif_level` to a value much smaller than 0.01, then your results may be quite different than what I have observed (e.g., see [this figure](https://github.com/DIDSR/ThresholdoutAUC/blob/master/visualizations/img/naive_holdout_reuse_vs_thresholdout_AUC_landscape.pdf), and other figures)
